symengine/functions.cpp

```diff
@@ -84,7 +84,7 @@ bool get_pi_shift(const RCP<const Basic> &arg,
         else if (size == 1) {
             // arg should be of form `a + n*pi/12`
             // where `a` is a `Number`.
-            auto p = s.dict_.begin();
+            auto p = s.get_dict_().begin();
```

Comment:
Segfault is because `s.get_dict_()` returns a copy of the dict and it is not stored. Only the iterator is stored and when the dictionary is cleared, `p` becomes invalid.